app/gui/view/execution-mode-selector/src/lib.rs

```diff
@@ -0,0 +1,250 @@
+        background.set_size(Vector2::new(max_width, HEIGHT));
+    }
+
+    fn init_play_icon(max_width: f32, shape: &play_icon::View) {
+        shape.set_size(Vector2::new(10.0, 11.0));
+        shape.set_x(max_width / 2.0 - PLAY_BUTTON_OFFSET);
+    }
+
+    fn update_position(&self, camera: &Camera2d) {
+        let screen = camera.screen();
+        let x = -screen.width / 2.0 + OVERALL_WIDTH / 2.0;
+        let y = screen.height / 2.0 - HEIGHT / 2.0;
+        self.inner_root.set_x(x.round());
+        self.inner_root.set_y(y.round());
+        error!("update_position x: {}, y: {}", x, y);
+    }
+
+    fn set_entries(&self, entries: Rc<Vec<ExecutionMode>>) {
+        let provider = ensogl_list_view::entry::AnyModelProvider::from(entries.clone_ref());
+        self.dropdown.set_entries(provider);
+        self.dropdown.set_selected(0);
+    }
+}
+
+impl display::Object for Model {
+    fn display_object(&self) -> &display::object::Instance {
+        &self.display_object
+    }
+}
+
+
+// =============================
+// === ExecutionModeDropdown ===
+// =============================
+
+impl component::Model for Model {
+    fn label() -> &'static str {
+        "ExecutionModeDropdown"
+    }
+
+    fn new(app: &Application) -> Self {
+        let scene = &app.display.default_scene;
+
+        let display_object = display::object::Instance::new();
+        let inner_root = display::object::Instance::new();
+        let background = background::View::new();
+        let play_icon = play_icon::View::new();
+        let dropdown = ensogl_drop_down_menu::DropDownMenu::new(app);
+
+        display_object.add_child(&inner_root);
+        inner_root.add_child(&dropdown);
+        inner_root.add_child(&play_icon);
+        inner_root.add_child(&background);
+
+        scene.layers.panel.add(&inner_root);
+        scene.layers.panel.add(&dropdown);
+        scene.layers.panel.add(&play_icon);
+
+        dropdown.set_label_layer(&scene.layers.panel_text);
+
+        Self::init_dropdown(&dropdown);
+        Self::init_background(OVERALL_WIDTH, &background);
+        Self::init_play_icon(OVERALL_WIDTH, &play_icon);
+
+        Self { display_object, background, play_icon, dropdown, inner_root }
+    }
+}
+
+impl component::Frp<Model> for Frp {
+    fn init(
+        network: &enso_frp::Network,
+        frp: &<Self as ensogl::application::frp::API>::Private,
+        app: &Application,
+        model: &Model,
+        _style_watch: &ensogl::display::shape::StyleWatchFrp,
+    ) {
+        let scene = &app.display.default_scene;
+        let camera = scene.camera();
+        let dropdown = &model.dropdown;
+        let play_icon = &model.play_icon;
+        let input = &frp.input;
+        let output = &frp.output;
+
+        frp::extend! { network
+            init <- source();
+
+            // Layout
+            let camera_changed = scene.frp.camera_changed.clone_ref();
+            update_position <- any(init, camera_changed);
```

Comment:
Why are we making position update on camera change? Instead, this should be added to al ayer with camera that does not move with navigator.

Reply:
This update is not about camera movement, but the viewport size change. As we want this to be aligned with the top of the window. This seems to be what all the components placed this way are using at the moment.

The alternative would be a new layer with a fixed origin at the top left window corner, but that would also not work for things placed at the bottom. We could fix the coordinates of that layer to be 0..1, but then finding the correct position for an object would be an issue as it would depend on the scaling. Or am I missing something?

Reply:
Hmm, we shouldd probably have a layer aligned to top left corner and place such items there. Items aligned to bottom left one would use another layer then. I mean, it should not be the responsibility of elements to place themselves. Nevertheless, this is a broader discussion and im ok with not fixing it here.